mopro-ffi/src/app_config/mod.rs

```diff
@@ -1,73 +1,74 @@
-use std::fs;
-use std::path::{Path, PathBuf};
-use std::process::Command;
-use uuid::Uuid;
+use camino::Utf8Path;
+use uniffi_bindgen::bindings::{KotlinBindingGenerator, SwiftBindingGenerator};
+use uniffi_bindgen::library_mode::generate_bindings;
 
-pub mod android;
-pub mod ios;
+use crate::app_config::utils::build_release;
 
-pub fn mktemp() -> PathBuf {
-    let dir = std::env::temp_dir().join(Path::new(&Uuid::new_v4().to_string()));
-    fs::create_dir(&dir).expect("Failed to create tmpdir");
-    dir
-}
+mod android;
+mod ios;
+mod utils;
 
-fn tmp_local(build_path: &Path) -> PathBuf {
-    let tmp_path = build_path.join("tmp");
-    if let Ok(metadata) = fs::metadata(&tmp_path) {
-        if !metadata.is_dir() {
-            panic!("non-directory tmp");
-        }
-    } else {
-        fs::create_dir_all(&tmp_path).expect("Failed to create local tmpdir");
-    }
-    tmp_path
-}
+use utils::*;
 
-pub fn mktemp_local(build_path: &Path) -> PathBuf {
-    let dir = tmp_local(build_path).join(&Uuid::new_v4().to_string());
-    fs::create_dir(&dir).expect("Failed to create tmpdir");
-    dir
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Target {
+    Android,
+    Ios,
 }
 
-pub fn cleanup_tmp_local(build_path: &Path) {
-    fs::remove_dir_all(tmp_local(build_path)).expect("Failed to remove tmpdir");
+#[derive(Debug, thiserror::Error)]
+pub enum MoproBuildError {
+    #[error("Failed to build the release library: {0}")]
+    LibraryBuildError(String),
+    #[error("Failed to generate bindings: {0}")]
+    GenerateBindingsError(String),
 }
 
-pub const UDL: &str = include_str!("../mopro.udl");
+pub fn build(target: Target) -> Result<(), MoproBuildError> {
+    // Build the crate as a release library for the bindgen
+    build_release().map_err(|e| MoproBuildError::LibraryBuildError(e.to_string()))?;
```

Comment:
It does save around 200seconds on my machine, will switch to debug

Reply:
I have actually followed your suggestion from above and instead of doing a separate compilation for `dylib` I am just reusing the result of `"aarch64-apple-ios"`. Thank you for the suggestion! Now it should be about the same compilation time as before.